### PR TITLE
Data Pipeline V2: Cleanup

### PR DESCRIPTION
**Pull request by tchaton**
## What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? [not needed for typos/docs]
- [ ] Did you verify new and existing tests pass locally with your changes?
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)?

## PR review
 - [x] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: flash/graph/classification/data.py [hunk 1]
```diff
@@ -94,3 +98,11 @@ def num_features(self):
         n_cls_val = getattr(self.val_dataset, "num_features", None)
         n_cls_test = getattr(self.test_dataset, "num_features", None)
         return n_cls_train or n_cls_val or n_cls_test
+
+    from_folders = None
```
Comment (tchaton) [inline flash/graph/classification/data.py:102]: @ethanwharris Do we want to do this for now? Just an idea, but might not be worth it.

File: flash/image/detection/data.py [hunk 2]
```diff
@@ -407,3 +412,80 @@ def from_fiftyone(
             ),
             **data_module_kwargs,
         )
+
+    @classmethod
+    def from_folders(
```
Comment (tchaton) [inline flash/image/detection/data.py:417]: @ethanwharris Not sure about those.

File: flash/image/detection/data.py [hunk 3]
```diff
@@ -407,3 +412,80 @@ def from_fiftyone(
...
+        cls,
+        predict_folder: Optional[str] = None,
+        predict_transform: Optional[Dict[str, Callable]] = None,
+        image_size: Tuple[int, int] = (128, 128),
+        **data_module_kwargs: Any,
+    ) -> "DataModule":
+        """Creates a :class:`~flash.core.data.data_module.DataModule` object from the given folders using the
+        :class:`~flash.core.data.io.input.Input` of name
+        :attr:`~flash.core.data.io.input.InputFormat.FOLDERS`
+        from the passed or constructed :class:`~flash.core.data.io.input_transform.InputTransform`.
```
Comment (ethanwharris) [inline flash/image/detection/data.py:427]: Would maybe delete this docstring as it is out of date

File: flash/image/detection/data.py (hunk 2 above)
Comment (ethanwharris) [reply flash/image/detection/data.py:417]: I like it :smiley:

File: flash/image/detection/data.py [hunk 4]
```diff
@@ -407,3 +412,80 @@ def from_fiftyone(
...
+        return cls(
+            None,
+            None,
+            None,
+            IceVisionInput(RunningStage.PREDICTING, predict_folder),
+            input_transform=cls.input_transform_cls(
+                None,
+                None,
+                None,
+                predict_transform,
```
Comment (ethanwharris) [inline flash/image/detection/data.py:446]: Could use kwargs here:
```suggestion
            predict_dataset=IceVisionInput(RunningStage.PREDICTING, predict_folder),
            input_transform=cls.input_transform_cls(
                predict_transform=predict_transform,
```

File: flash/image/detection/data.py [hunk 5]
```diff
@@ -407,3 +412,80 @@ def from_fiftyone(
...
+        cls,
+        predict_files: Optional[List[str]] = None,
+        predict_transform: Optional[Dict[str, Callable]] = None,
+        image_size: Tuple[int, int] = (128, 128),
+        **data_module_kwargs: Any,
+    ) -> "DataModule":
+        """Creates a :class:`~flash.core.data.data_module.DataModule` object from the given folders using the
+        :class:`~flash.core.data.io.input.Input` of name
+        :attr:`~flash.core.data.io.input.InputFormat.FOLDERS`
+        from the passed or constructed :class:`~flash.core.data.io.input_transform.InputTransform`.
```
Comment (ethanwharris) [inline flash/image/detection/data.py:463]: Same comment as for docstring above

File: flash/image/detection/data.py [hunk 6]
```diff
@@ -407,3 +412,80 @@ def from_fiftyone(
...
+        """
+        return cls(
+            None,
+            None,
+            None,
+            IceVisionInput(RunningStage.PREDICTING, predict_files),
+            input_transform=cls.input_transform_cls(
+                None,
+                None,
+                None,
```
Comment (ethanwharris) [inline flash/image/detection/data.py:481]: Could also use kwargs here

**Review by ethanwharris — APPROVED** [review]
LGTM, some small comments. @tchaton I think you will also need to update the instance segmentation and keypoint detection (maybe object detection too) examples that use the current `from_folders` method to use the `from_icedata` method instead.

File: flash/image/detection/data.py (hunk 3 above)
Comment (tchaton) [reply flash/image/detection/data.py:427]: Done !

File: flash/image/detection/data.py (hunk 4 above)
Comment (tchaton) [reply flash/image/detection/data.py:446]: Done !

File: flash/image/detection/data.py (hunk 5 above)
Comment (tchaton) [reply flash/image/detection/data.py:463]: Done !

File: flash/image/detection/data.py (hunk 6 above)
Comment (tchaton) [reply flash/image/detection/data.py:481]: Done !

**Pull request merged**